_posts/2021-02-26-dune-3-plan.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: blog
+title: Plan for Dune 3.0
+author: andreymokhov
+tags: [dune]
+---
+
+<!-- TODO: After publishing the post, create a Discuss page, and link to it. -->
+
+The Dune team has started planning the next major release and we would like to
+share and discuss the current plan with the community. We expect to release
+Dune 3 in August 2021. As with previous major releases, we will put a lot of
+effort in backward compatibility, which means the majority of Dune users should
```

Comment:
I think that's always a good opportunity to explain the dune version vs dune-lang version - upgrading dune itself is still a safe operation as it will not change how existing projects are built. Some project-related breaking changes will happen (e.g. scoping semantics) but these will only happen when you as a maintainer flip the switch with `(lang dune 3.0)`.

Reply:
Thanks! I agree, it's good to remind about the difference. However, in the specific case of scoping semantics, we do plan to change the behaviour unconditionally, because otherwise it would be really difficult to reason about nested projects that specify different version of the Dune language. It seems worth clarifying this in the text.

Reply:
Absolutely -- otherwise there would be no point doing a major version bump if there was no way to break compat.

It might be worth having a short section explicitly about forward and backwards compatibility -- just today on the opam meeting it emerged that the opam developers didn't know about the "dune-lang" _vs_ "dune" distinction and were working around a dune bug as a result.  The post about dune 3 is a good chance to set expectations: users can continue to use dune-lang in dune 3.0, _except_ for the planned breaking changes around scoping (which are important to continue to scale the monorepo approach).

Reply:
I've updated the paragraph in the intro, trying to keep it brief. Hope it makes things clearer.

It seems useful to have a separate page going into more details about the Dune vs Dune language distinction, which we can link to from various places (including such blog posts).

Reply:
I made another pass and a separate section on breaking changes. Please have a look.